vmdk_plugin/drivers/vmdk/vmdk_driver.go

```diff
@@ -298,50 +296,71 @@ func (d *VolumeDriver) processMount(r volume.MountRequest) volume.Response {
+		r.Options["fstype"] = fs.FstypeDefault
+	}
+}
+
+// cloneFrom clones an existing volume.
+func (d *VolumeDriver) cloneFrom(r volume.Request) volume.Response {
+	errClone := d.ops.Create(r.Name, r.Options)
+	if errClone != nil {
+		log.WithFields(log.Fields{"name": r.Name, "error": errClone}).Error("Clone volume failed ")
+		return volume.Response{Err: errClone.Error()}
+	}
+	return volume.Response{Err: ""}
+}
+
+// detach detaches a volume, or prints a warning log on failure.
+func (d *VolumeDriver) detach(name string, opts map[string]string) {
+	errDetach := d.ops.Detach(name, opts)
+	if errDetach != nil {
+		log.WithFields(log.Fields{"name": name, "error": errDetach}).Warning("Detach volume failed ")
+	}
+}
+
+// remove removes a volume, or prints a warning log on failure.
+func (d *VolumeDriver) remove(name string, opts map[string]string) {
+	errRemove := d.ops.Remove(name, opts)
+	if errRemove != nil {
+		log.WithFields(log.Fields{"name": name, "error": errRemove}).Warning("Remove volume failed ")
+	}
+}
+
+// forceDetachAndRemove detaches a volume and then removes it, ignoring any errors.
+func (d *VolumeDriver) forceDetachAndRemove(name string, opts map[string]string) {
+	d.detach(name, opts)
+	d.remove(name, opts)
+}
+
 // No need to actually manifest the volume on the filesystem yet
 // (until Mount is called).
 // Name and driver specific options passed through to the ESX host
 
-// Create - create a volume.
+// Create creates a volume.
 func (d *VolumeDriver) Create(r volume.Request) volume.Response {
+	d.prepareCreateOptions(r)
 
-	if r.Options == nil {
-		r.Options = make(map[string]string)
-	}
 	// If cloning a existent volume, create and return
-	if _, result := r.Options["clone-from"]; result == true {
-		errClone := d.ops.Create(r.Name, r.Options)
-		if errClone != nil {
-			log.WithFields(log.Fields{"name": r.Name, "error": errClone}).Error("Clone volume failed ")
-			return volume.Response{Err: errClone.Error()}
-		}
-		return volume.Response{Err: ""}
-	}
-
-	// Use default fstype if not specified
-	if _, result := r.Options["fstype"]; result == false {
-		r.Options["fstype"] = fs.FstypeDefault
+	if _, result := r.Options["clone-from"]; result {
+		return d.cloneFrom(r)
 	}
 
-	// Get existent filesystem tools
-	supportedFs := fs.MkfsLookup()
-
-	// Verify the existence of fstype mkfs
-	mkfscmd, result := supportedFs[r.Options["fstype"]]
-	if result == false {
-		msg := "Not found mkfs for " + r.Options["fstype"]
-		msg += "\nSupported filesystems found: "
-		validfs := ""
-		for fs := range supportedFs {
-			if validfs != "" {
-				validfs += ", " + fs
-			} else {
-				validfs += fs
-			}
-		}
-		log.WithFields(log.Fields{"name": r.Name,
-			"fstype": r.Options["fstype"]}).Error("Not found ")
-		return volume.Response{Err: msg + validfs}
+	// Check whether the fstype filesystem is supported.
+	errFstype := fs.VerifyFSSupport(r.Options["fstype"])
```

Comment:
[prepareCreateOptions()](https://github.com/vmware/docker-volume-vsphere/pull/1504/files/60618d43aec4c9cdc406df58d050b0f4c7711862#diff-aeea88b6324054966e71ce0340771803R352) sets `fstype` if not already set.